core/src/main/java/net/neoforged/fml/config/ModConfig.java

```diff
@@ -24,17 +23,20 @@ public class ModConfig
     private final ModContainer container;
     private final ConfigFileTypeHandler configHandler;
     private CommentedConfig configData;
-    private Callable<Void> saveHandler;
 
-    public ModConfig(final Type type, final IConfigSpec<?> spec, final ModContainer container, final String fileName) {
+    public ModConfig(final Type type, final IConfigSpec<?> spec, final ModContainer container, final String fileName, ConfigFileTypeHandler configHandler) {
```

Comment:
Why a custom handler? Is this to implement a custom format? (Not sure we want that?)

Reply:
I had a choice between deleting the `configHandler` field or creating it via constructor, which is what I chose, but yes, custom formats are useless since toml is fine

Reply:
I don't support this change.
Fml solely supports toml configs and this opens the door to others, which fractures the eco system for no good reason. Please revert. The original code with setting the field in the constructor is fit for purpose.

Hell I would even argue the other way around and remove the field entirely